src/RackspaceAdapter.php

```diff
@@ -158,7 +159,20 @@ public function createDir($dirname, Config $config)
      */
     public function writeStream($path, $resource, Config $config)
     {
-        return $this->write($path, $resource, $config);
+        $location = $this->applyPathPrefix($path);
+        $headers = [];
+
+        if ($config && $config->has('headers')) {
+            $headers = $config->get('headers');
+        }
```

Comment:
`$config` is not nullable, so I would not worry about testing it. Also `$config->get(` accepts to parameters, the second one being the default value, so this can be simplified a lot:

```suggestion
        $headers = $config->get('headers', []);
```

Reply:
Updated